core/modules/shrscan/Director.h

```diff
@@ -0,0 +1,315 @@
+// -*- LSST-C++ -*-
+/*
+ * LSST Data Management System
+ * Copyright 2014-2015 LSST Corporation.
+ *
+ * This product includes software developed by the
+ * LSST Project (http://www.lsst.org/).
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the LSST License Statement and
+ * the GNU General Public License along with this program.  If not,
+ * see <http://www.lsstcorp.org/LegalNotices/>.
+ */
+
+#ifndef LSST_QSERV_SSCAN_DIRECTOR_H
+#define LSST_QSERV_SSCAN_DIRECTOR_H
+
+// System headers
+#include <memory>
+#include <string>
+#include <vector>
+
+namespace lsst {
+namespace qserv {
+namespace shrscan {
+
+//-----------------------------------------------------------------------------
+//! @brief Describe a required table for a shared scan query.
+//-----------------------------------------------------------------------------
+
+class TableInfo
+{
+public:
+
+std::string tableName;    //< Name of the table
+bool        lockIndex;    //< If true, the table's index is locked, if any.
+bool        lockData;     //< If true, the table's data  is locked
+
+//-----------------------------------------------------------------------------
+//! Constructor
+//!
+//! @param  tabName   is the name of the table.
+//! @param  lkIndex   whether or not to lock the table's index
+//! @param  lkData    whether or not to lock the table's data
+//-----------------------------------------------------------------------------
+
+            TableInfo(const char *tabName="", bool lkIndex=true, bool lkData=true)
+                     : tableName(tabName), lockIndex(lkIndex), lockData(lkData)
+                     {}
+
+//-----------------------------------------------------------------------------
+//! Destructor
+//-----------------------------------------------------------------------------
+
+           ~TableInfo() {}
```

Comment:
Well, I probably read too much "modern C++" stuff, but anyways. With value-like types (non-abstract classes that can be copied/assigned) you only need to declare non-default destructor if the class has some non-trivial resources (pointers, etc.). But in those cases one also has to define non-trivial copy constructor and assignment operator. Basically rule of thumb is that if you need one then you need all three, having only one of them declared raises suspicion that something may be forgotten. Personally in general I prefer not to write code that I don't have to write (unless I'm paid by LOC count :smile: ), and reading code that does nothing is a waste of time too (just as writing long comments about it :angel: ).